cli/internal/renderer/graph/utils.go

```diff
@@ -0,0 +1,55 @@
+package graph
+
+import (
+	"cmp"
+	"context"
+	"slices"
+
+	syncdag "ocm.software/open-component-model/bindings/go/dag/sync"
+)
+
+// GetNeighborsSorted returns the neighbors of the given vertex sorted by their
+// order index if available, otherwise by their key.
+// This function may be used to implement Renderer with a consistent
+// order of neighbors in the output.
+func GetNeighborsSorted[T cmp.Ordered](ctx context.Context, vertex *syncdag.Vertex[T]) []T {
+	var neighbors []T
+
+	vertex.Edges.Range(func(key, _ any) bool {
+		if childId, ok := key.(T); ok {
+			neighbors = append(neighbors, childId)
+		}
+		return true
+	})
+
+	slices.SortFunc(neighbors, func(a, b T) int {
+		return compareByOrderIndex(ctx, vertex, a, b)
+	})
+
+	return neighbors
+}
+
+func compareByOrderIndex[T cmp.Ordered](ctx context.Context, vertex *syncdag.Vertex[T], a, b T) int {
+	orderA := getOrderIndex(ctx, vertex, a)
+	orderB := getOrderIndex(ctx, vertex, b)
+
+	// If both edges have order indices, compare them.
+	if orderA != nil && orderB != nil {
+		return *orderA - *orderB
```

Comment:
if both orderA and orderB are set, isn't it wrong to subtract? instead of calling

```suggestion
            return cmp.Compare(*orderA, *orderB)
```

or even

```suggestion
            a, b = *orderA, *orderB
```